### PR TITLE
Update Resource Management samples to use stable version of Windows App SDK

### PR DESCRIPTION
**Pull request by erik0686**
## Description

This updates the Resource Management samples to use the latest stable Windows App SDK nuget package. This also required updating the SDK BuildTools.
This also removes the previous way of loading the Windows App SDK and replaces it with the current way of adding 
<WindowsPackageType>None</WindowsPackageType>

Verified by building in Release and Debug in ARM64, x86 and x64. And by running them in x86 and x64, since I dont have an ARM device

--Note: the cs-winui sample update will come on a different PR, as Rohan is working on it :) 

## Checklist

- [x] Samples build and run using the Visual Studio versions listed in the [Windows development docs](https://docs.microsoft.com/windows/apps/windows-app-sdk/set-up-your-development-environment?tabs=stable#2-install-visual-studio).
- [x] Samples build and run on all supported platforms (x64, x86, ARM64) and configurations (Debug, Release). 
- [x] Samples set the minimum supported OS version to Windows 10 version 1809.
- [x] Samples build clean with no warnings or errors.
- [ ] **[For new samples]**: Samples have completed the [sample guidelines checklist](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#checklist) and follow [standardization/naming guidelines](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#standardization-and-naming).

### CODE REVIEW

File: Samples/ResourceManagement/cpp-winui/winui_desktop_packaged_app_cpp/packages.config [hunk 1]
```diff
@@ -2,6 +2,6 @@
 <packages>
   <package id="Microsoft.VCRTForwarders.140" version="1.0.7" targetFramework="native" />
   <package id="Microsoft.Windows.CppWinRT" version="2.0.210714.1" targetFramework="native" />
-  <package id="Microsoft.Windows.SDK.BuildTools" version="10.0.20348.19" targetFramework="native" />
-  <package id="Microsoft.WindowsAppSDK" version="1.0.0-preview1" targetFramework="native" />
+  <package id="Microsoft.Windows.SDK.BuildTools" version="10.0.22499-preview" targetFramework="native" />
+  <package id="Microsoft.WindowsAppSDK" version="1.0.0-20211111.1-CI-stable" targetFramework="native" />
```
Comment (angelazhangmsft) [inline Samples/ResourceManagement/cpp-winui/winui_desktop_packaged_app_cpp/packages.config:6]: I assume we will wait for the 1.0.0 release and update these versions to "1.0.0"? This version is internal only

File: Samples/ResourceManagement/cpp-winui/winui_desktop_packaged_app_cpp/packages.config [hunk 2]
```diff
@@ -2,6 +2,6 @@
 <packages>
   <package id="Microsoft.VCRTForwarders.140" version="1.0.7" targetFramework="native" />
   <package id="Microsoft.Windows.CppWinRT" version="2.0.210714.1" targetFramework="native" />
-  <package id="Microsoft.Windows.SDK.BuildTools" version="10.0.20348.19" targetFramework="native" />
-  <package id="Microsoft.WindowsAppSDK" version="1.0.0-preview1" targetFramework="native" />
+  <package id="Microsoft.Windows.SDK.BuildTools" version="10.0.22499-preview" targetFramework="native" />
```
Comment (angelazhangmsft) [inline Samples/ResourceManagement/cpp-winui/winui_desktop_packaged_app_cpp/packages.config:5]: Should we be using a preview version of this package?

File: Samples/ResourceManagement/cpp-winui/winui_desktop_packaged_app_cpp/packages.config (hunk 1 above)
Comment (erik0686) [reply Samples/ResourceManagement/cpp-winui/winui_desktop_packaged_app_cpp/packages.config:6]: Good point Angela :)

File: Samples/ResourceManagement/cs-winforms-unpackaged/winforms_unpackaged_app.csproj [hunk 3]
```diff
@@ -5,13 +5,14 @@
     <TargetFramework>net5.0-windows10.0.19041.0</TargetFramework>
     <TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
     <UseWindowsForms>true</UseWindowsForms>
+    <WindowsPackageType>None</WindowsPackageType>
```
Comment (rohanp-msft) [inline Samples/ResourceManagement/cs-winforms-unpackaged/winforms_unpackaged_app.csproj:8]: ><WindowsPackageType>None</WindowsPackageType> [](http://example.com/codeflow?start=4&length=45)

curious about this. is this necessary now?

File: Samples/ResourceManagement/cs-wpf/wpf_packaged_app/wpf_packaged_app.csproj [hunk 4]
```diff
@@ -10,8 +10,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.20348.19" />
-    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.0-preview1" />
+    <PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22499-preview" />
```
Comment (rohanp-msft) [inline Samples/ResourceManagement/cs-wpf/wpf_packaged_app/wpf_packaged_app.csproj:13]: >10.0.22499-preview [](http://example.com/codeflow?start=74&length=18)

was this necessary, Erik? the upgrade, i mean

File: Samples/ResourceManagement/cs-winforms-unpackaged/winforms_unpackaged_app.csproj (hunk 3 above)
Comment (erik0686) [reply Samples/ResourceManagement/cs-winforms-unpackaged/winforms_unpackaged_app.csproj:8]: Yeah :) As of preview3, this is the new way to load the dependency we needed. As I understand, having this property makes nuget do the rest for you by using module initializers behind the scenes. If the initializer is simple, you can just do it with the tag, or you can keep the method and customize it (which we don't need). In fact, we don't even need the MddBootstrap.cs file anymore, which I forgot to remove. I'll do that as well :)

File: Samples/ResourceManagement/cs-wpf/wpf_packaged_app/wpf_packaged_app.csproj (hunk 4 above)
Comment (erik0686) [reply Samples/ResourceManagement/cs-wpf/wpf_packaged_app/wpf_packaged_app.csproj:13]: Yes and no. It is necessary because it seems the WAS stable release has the SDK.BuildTools with a minimal version of 10.0.22000 as dependency. 
And no because I think we might not want to use the preview version, so I I should probably change that to 22000 :)

File: Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj [hunk 5]
```diff
@@ -34,9 +34,11 @@
     <VCProjectVersion>16.0</VCProjectVersion>
     <Keyword>Win32Proj</Keyword>
     <ProjectGuid>{fd5c9f04-43c5-4a1c-b3f2-f6725a4ca4ac}</ProjectGuid>
+    <WindowsPackageType>None</WindowsPackageType>
     <RootNamespace>consoleunpackagedapp</RootNamespace>
     <WindowsTargetPlatformVersion>10.0.19041.0</WindowsTargetPlatformVersion>
     <WindowsTargetPlatformMinVersion>10.0.17763.0</WindowsTargetPlatformMinVersion>
+    <LogSDKReferenceResolutionErrorsAsWarnings>true</LogSDKReferenceResolutionErrorsAsWarnings>
```
Comment (angelazhangmsft) [inline Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj:41]: is this addition needed for update to 1.0?

**Review by angelazhangmsft — APPROVED** [review]

File: Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj (hunk 5 above)
Comment (erik0686) [reply Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj:41]: Removing this generates WAS known errors for which this is an also known workaround. I know this has been used across several Microsoft projects such as WinUI.

**Review by rohanp-msft — APPROVED** [review]
